extensions/nba-game-viewer/package.json

```diff
@@ -18,43 +18,67 @@
     {
       "name": "upcoming-games",
       "title": "View Upcoming Games",
-      "subtitle": "NBA",
-      "description": "Fetches upcoming NBA games",
+      "subtitle": "NBA & WNBA",
+      "description": "Fetches upcoming NBA or WNBA games",
       "mode": "view"
     },
     {
       "name": "standings",
       "title": "View Standings",
-      "subtitle": "NBA",
-      "description": "Fetches NBA standings",
+      "subtitle": "NBA & WNBA",
+      "description": "Fetches NBA or WNBA standings",
       "mode": "view"
     },
     {
       "name": "headlines",
       "title": "View Headlines",
-      "subtitle": "NBA",
-      "description": "Fetches NBA league headlines from ESPN",
+      "subtitle": "NBA & WNBA",
+      "description": "Fetches NBA or WNBA league headlines from ESPN",
       "mode": "view"
     }
   ],
-  "preferences": [{
-    "name": "conferences",
-    "description": "Choose Standings default conference",
-    "type": "dropdown",
-    "title": "Conferences",
-    "data": [{
-        "title": "Eastern Conference",
-        "value": "Eastern"
-      }, {
-        "title": "Western Conference",
-        "value": "Western"
-      }, {
-        "title": "League",
-        "value": "League"
-    }],
-    "required": true,
-    "default": "Eastern"
-  }],
+  "preferences": [
```

Comment:
I am always not to add more config. The only reason I added the preference was the differences in API paths, etc. I didn't know about `storeValue`. I am happy to push a new commit providing that capability in the next couple of days.

Reply:
For sure, this suggestion is non-blocking for the PR, but I believe it will be a better/streamlined UX for users. If you want to address this in a followup PR, feel free to resolve this comment.

Reply:
How do you feel about removing the conference preference, too? And just relying on the dropdown?

Reply:
Wait, the `storeValue`  will always stay to the last selected value ? 👀

I like to open the View Standing to League, then I select East or West but when I come back later it's always League 🙏🏾 

Preference means Preference, you only touch it once... Maybe offers a `none` option

Reply:
@degouville The current default is Eastern - maybe I misremember it, but it always opens to Eastern. I can change it but when I re-open it's always back at Eastern.

Reply:
@degouville The problem is the league - if I specify Eastern - do I default to NBA Eastern? Or WNBA Eastern? Since the preference was to not have a league preference I am not sure how to solve for this.

Reply:
> @degouville do I default to NBA Eastern? Or WNBA Eastern? 

IMO NBA & WNBA are 2 different leagues.

Maybe having an Opt IN/OUT preference to 
- Have NBA but no WNBA
- Have WNBA but no NBA
- Have both for those who are interested

And then display the dropdown options accordingly?

Reply:
Okay - this is my compromise in the latest commit:

If the user has a preference for League or Eastern or Western conferences, then they are open to that conference in the previous league they had open, either NBA or WNBA. If no league is stored, then the standings will default to the NBA and their conference preference.

Reply:
> If no league is stored, then the standings will default to the NBA and their conference preference.

But if I opened the Standings, arrived in *League*, moved to *Western* then pressed `ESC`
...then I come back letter it will be *Western* right ? 🤔🤔

Is it at least possible to have an Opt In pref to *add* this behaviour?

Having added option to get WNBA data is cool for WNBA fans but hope we dont change the actual behavior of NBA Standings now

Reply:
Ok I think there is some confusion here. Removing the preference doesn't mean we are removing any option for user to select NBA/WNBA eastern/western/league. And if we are advocating for fast switching, accessory is a better tool rather than preferences. Preferences are meant to be touched very less frequently rather than accessory dropdown. There is always a way to present dropdown like this:

* section - NBA
 * item - Eastern Conference
 * item - Western Conference
 * item - League
* section - WNBA
 * item - Eastern Conference
 * item - Western Conference
 * item - League

And if we use `storeValue` then it essentially becomes a preference, w/o the user working with two raycast windows (command + preference) to get what they want. That's what I meant by saying preference is doing the same thing as dropdown, but with extra steps. cc: @jamtur01 @degouville

Reply:
@degouville No, it'll be League.

Reply:
> How do you feel about removing the conference preference, too? And just relying on the dropdown?

Yeah I don't think we need any preference for these commands, b/c everything can be essentially handled by drop-down.

Reply:
I am a WNBA fan; I have no interest in the NBA, but they are very similar if not nearly identical, leagues in most people's eyes. So, I made the existing extension work for both groups because that made the most sense to me. I am happy to write another extension for the WNBA or to add preferences to select the league as I initially did. As far as I can see, the proposed compromise changes nothing for the NBA standings. Please feel free to test the extension and confirm.

Reply:
> The conference will return to the preference in the last league viewed.

Sorry @jamtur01 that's exactly what I'd like to avoid.

If I set League in my Preferences, I want to see League every time I open Standings. If I switch to E or W then I close, while I'm back it should be League

BUT

**_If there's nothing set in Preferences_**, feel free to return to the last viewed selection everytime you call the Standings

Reply:
> we need any preference for these commands

I need it @jfkisafk 😊🙏🏾

NGL I became used to Raycast in the 1st place for this one little feature lol

Reply:
@degouville I think we're talking at cross-purposes - the extension will behave precisely as you've requested.

1. If you have League in the preference and open it to NBA and then close it - it'll return to NBA League.
2. If you have League in the preferences and open it to WNBA and then close it - it’ll return to WNBA League.
3. If you have Western in the preferences, open it to NBA and then close it, and it'll return to NBA League.
4. If you have NO preference set, it'll go to NBA Eastern, as that's the default already.

The change I am proposing does not change the existing behaviour EXCEPT it returns to the last league you viewed: the NBA or the WNBA.

Reply:
> I am happy to add preferences to select the league 

Let's try this

NBA & WNBA both OptIn by default AND no preference for the League/Conference to display
then I'll see the 6 options: Both Leagues + 2 Eastern & 2 Western in the Dropdown with last stored value

IF I OptOut the WNBN in preference AND set preferred Standing to League
then I'll see only 3 options: NBA League + 1 Eastern & 1 Western and always the preferred without the last stored value

what do you think @jamtur01 @jfkisafk  ?

Reply:
Honestly, this is a little frustrating. That makes it more complex and adds considerably more logic and preferences. The feedback I was given was not to do that. I invested time in providing what is a valuable addition to the extension with minimal to no impact on the existing extension. The only change is seeing both leagues in a dropdown. And I do not understand why seeing the WNBA and NBA conferences is such a big deal, especially when the behavior in the existing standings DOES NOT CHANGE. I am happy to maintain a fork and close this PR if this is an issue.

Reply:
> EXCEPT it returns to the last league you viewed: the NBA or the WNBA.

I'm happy with it. TY @jamtur01 ❣️

I also think most could benefit having the WNBA on another view to be able to set differents Hotkeys, but that's just a suggestions. Thank's for your time explaining the whole thing 👌🏾

Reply:
+1 to what @jamtur01 suggested. I have tested his changes and everything works except for the whole conference default value coming from preferences. I am putting together some changes on his branch (I will produce a gist and a demo and upload it here).

Reply:
**1st I commented because @jfkisafk suggestion to remove all the preference just to deal with stored values**

> The only change is seeing both leagues in a dropdown. And I do not understand why seeing the WNBA and NBA leagues and their conferences is such a big deal

Imagine some friends, confortable on a sofa but the keyboard not in a perfect ergonomic setup, after 3 beers joking and trashtalking each other regarding the rankings. Not paying attention to every clicks, the point is to laugh & have fun. One setted an Hotkey to open the standings always on League. 
Maybe they don't want to accidentally fall on WNBA results they don't necessarily care about because they pressed 3 times on the down arrow key instead of twice, then having to switch back every time...

Really glad that you add WNBA options. TY❤️ but try take everyone in consideration please.

Also, the other way, if one is just interested in WNBA but not NBA, it could be fair to be able to at least OptOut in preferences or call a dedicated view.

Reply:
That seems like quite a niche edge case. But clearly YMMV. Thanks for your feedback.

Reply:
Here is the gist of changes I was recommending initially @jamtur01 @degouville  : https://gist.github.com/jfkisafk/3e2883968d92bb03c4bfb562076fc2b7 Ofc, we can add icons to the league and conferences dropdown items for better UI but that's optional.

It will store the last value when the command is accessed and provide for eastern/western conference selection (and there are no preferences):
![2024-08-15 22 10 15](https://github.com/user-attachments/assets/d0aeb980-bf42-4e64-9f6f-a50a119e0993)

<img width="862" alt="Screenshot 2024-08-15 at 10 11 46 PM" src="https://github.com/user-attachments/assets/48a6296a-94b4-482f-8556-55a3c7291de5">

Reply:
> That seems like quite a niche edge case. But clearly YMMV. Thanks for your feedback.

Thank you for diminishing other's opinion. That's why we love open source 😄

Reply:
> Here is the gist of changes I was recommending initially

Works for me.

Reply:
For sure, then you can apply the fixes and maybe `npm run fix-lint` (to fix any lint issues) and resolve this thread. This is a 2-way door and we can always re-add preferences in case storeValue is sub-optimal for vast use cases of this command.

Reply:
Breaks the Preference system in place + Headlines are OK ATM

> Here is the gist of changes I was recommending initially @jamtur01 @degouville : https://gist.github.com/jfkisafk/3e2883968d92bb03c4bfb562076fc2b7 Ofc, we can add icons to the league and conferences dropdown items for better UI but that's optional.

> It will store the last value when the command is accessed and provide for eastern/western conference selection (and there are no preferences)


https://github.com/user-attachments/assets/4ca1fc8f-2944-44db-bf46-2650380ce1f5

Reply:
Oh sorry yeah I just demo'd the standings command. But essentially, the NBA/WNBA dropdown can be applied to "Upcoming Games" and "Headlines" commands as I initially suggested.

Reply:
> Oh sorry yeah I just demo'd the standings command. But essentially, the NBA/WNBA dropdown can be applied to "Upcoming Games" and "Headlines" commands as I initially suggested.

It's already applied in the current branch for the upcoming games and headlines. The only tab that has the conferences is the standings.